client/me/account/main.jsx

```diff
@@ -138,15 +145,29 @@ const Account = createReactClass( {
 	},
 
 	getUserSetting( settingName ) {
-		return this.props.userSettings.getSetting( settingName );
+		return (
+			get( this.props.unsavedUserSettings, settingName ) ??
```

Comment:
We're using nullish coalescing anyway to fall back to original setting, so should we avoid `get()` and use `this.props.unsavedUserSettings?.[ settingName ]` here instead?

Reply:
`settingName` could be a path so we need to fall back to using `get` as long as we don't have our own version of it.

Reply:
That said this could be a very good follow up. I already made a note to extract the recently introduced `setValue` and `removeValue` helpers as well (see https://github.com/Automattic/wp-calypso/pull/49521#discussion_r567798686).